scripts/jmwalletd.py

```diff
@@ -0,0 +1,626 @@
+        # self.cookie = encoded_token
+        self.cookie = encoded_token
+        #self.cookie = request.getHeader(b"JMCookie")
+
+
+        if self.cookie is None:
+            raise NotAuthorized("No cookie")
+
+        # the daemon blocks here until the wallet synchronization
+        # from the blockchain interface completes; currently this is
+        # fine as long as the client handles the response asynchronously:
+        self.wallet_service = WalletService(wallet)
+        while not self.wallet_service.synced:
+            self.wallet_service.sync_wallet(fast=True)
+        self.wallet_service.startService()
+        # now that the WalletService instance is active and ready to
+        # respond to requests, we return the status to the client:
+
+        #def response(request, succeed=True, status=200, **kwargs):
+        if('seedphrase' in kwargs):
+            return response(request,
+                        walletname=self.wallet_service.get_wallet_name(),
+                        already_loaded=False,token=encoded_token,seedphrase = kwargs.get('seedphrase'))
+        else:
+            return response(request,
+                        walletname=self.wallet_service.get_wallet_name(),
+                        already_loaded=False,token=encoded_token)
+
+    @app.route('/wallet/<string:walletname>/unlock', methods=['POST'])
+    def unlockwallet(self, request, walletname):
+        print_req(request)
+        #print(get_current_chain_params())
+        assert isinstance(request.content, BytesIO)
+        auth_json = self.get_POST_body(request, ["password"])
+        if not auth_json:
+            raise InvalidRequestFormat()
+        password = auth_json["password"]
+        if self.wallet_service is None:
+            wallet_path = get_wallet_path(walletname, None)
+            try:
+                wallet = open_test_wallet_maybe(
+                        wallet_path, walletname, 4,
+                        password=password.encode("utf-8"),
+                        ask_for_password=False)
+            except StoragePasswordError:
+                raise NotAuthorized("invalid password")
+            except StorageError as e:
+                # e.g. .lock file exists:
+                raise NotAuthorized(repr(e))
+            return self.initialize_wallet_service(request, wallet)
+        else:
+            print('wallet was already unlocked.')
+            return response(request,
+                            walletname=self.wallet_service.get_wallet_name(),
+                            already_loaded=True)
+
+
+    #This route should return list of current wallets created.
+    @app.route('/wallet/all', methods=['GET'])
+    def listwallets(self, request):
+        #this is according to the assumption that wallets are there in /.joinmarket by default, also currently path for linux system only.
+        #first user taken for path
+        user_path = glob.glob('/home/*/')[0]
```

Comment:
Use `jmbase.support.lookup_appdata_folder` for this (and you can remove the `glob` import).

Reply:
Oh sorry, I just remembered this is a slightly dumb comment :)

Basically, we load the `datadir` field in the global configuration on startup. By default it is loaded from `lookup_appdata_folder` but it can otherwise be set by the user with `--datadir=`. Either way, inside the script, you can access the correct value with `jm_single().datadir`. See, e.g.

https://github.com/JoinMarket-Org/joinmarket-clientserver/blob/cdd1a77d8553e9330c130d763da89f550ef96216/jmclient/jmclient/wallet_utils.py#L1451

Reply:
Fixed in https://github.com/JoinMarket-Org/joinmarket-clientserver/pull/996/commits/f8bb12a4421cd5114d4fdd4d7cba133f316ca118